credentials/sts/sts.go

```diff
@@ -192,15 +192,8 @@ type httpDoer interface {
 	Do(req *http.Request) (*http.Response, error)
 }
 
-func makeHTTPClient(roots *x509.CertPool) httpDoer {
-	return &http.Client{
-		Timeout: stsRequestTimeout,
-		Transport: &http.Transport{
-			TLSClientConfig: &tls.Config{
-				RootCAs: roots,
-			},
-		},
-	}
+func makeHTTPClient(client *http.Client) httpDoer {
```

Comment:
And for existing tests to continue to work, which override the `http.Client` with a fake one, the `httpDoer` interface needs to be exported as `HTTPDoer` and the field in the `Options` struct should take a `HTTPDoer` interface instead of a concrete `*http.Client`.

Reply:
i'm missing something there.   all the existing tests run as is  though

```
$ go test   -timeout 10s -run ^Test$ google.golang.org/grpc/credentials/sts
ok  	google.golang.org/grpc/credentials/sts	2.136s
```